### PR TITLE
Issues/4064 headless mode

### PR DESCRIPTION
**Pull request by orgonth**
Fix for #4064

Make CP work in headless mode:
- with `-i` option
- without `--first-image-set`

### CODE REVIEW

**Comment by 0x00b1** [comment]
Thanks, @orgonth !

File: cellprofiler/__main__.py [hunk 1]
```diff
@@ -766,7 +766,7 @@ def run_pipeline_headless(options, args):
         else:
             image_set_start = int(options.first_image_set)
     else:
-        image_set_start = None
+        image_set_start = 1
```
Comment (0x00b1) [inline cellprofiler/__main__.py:769]: @orgonth What's going on here?

File: cellprofiler/__main__.py [hunk 2]
```diff
@@ -838,17 +838,14 @@ def run_pipeline_headless(options, args):
     elif options.image_directory is not None:
         pathnames = []
 
-        os.path.walk(
-            os.path.abspath(options.image_directory),
-            lambda pathnames, dirname, fnames: pathnames.append(
+        for dirname, subdirs, fnames in os.walk(os.path.abspath(options.image_directory)):
```
Comment (0x00b1) [inline cellprofiler/__main__.py:841]: `subdirs` should be `_`

File: cellprofiler/__main__.py (hunk 1 above)
Comment (orgonth) [reply cellprofiler/__main__.py:769]: The method `Pipeline.run()` has a parameter `image_set_start`, which default value was changed from `None` to 1 in CP4.
However, in headless mode, `image_set_start` is always explicitly set when calling the method, based on the command-line parameter `--first-image-set`. I simply changed the value if that parameter is not given, to agree with the new default of `Pipeline.run()`.

File: cellprofiler/__main__.py (hunk 2 above)
Comment (orgonth) [reply cellprofiler/__main__.py]: Done.

**Comment by 0x00b1** [comment]
Thanks @orgonth !

**Pull request merged**